Kernel/Devices/USB/UHCIController.h

```diff
@@ -63,7 +74,28 @@ class UHCIController final : public PCI::Device {
 
     virtual void handle_irq(const RegisterState&) override;
 
+    void create_structures();
+    void setup_schedule();
+    QueueHead* allocate_queue_head() const;
+    TransferDescriptor* allocate_transfer_descriptor() const;
+
+private:
     IOAddress m_io_base;
+
+    Vector<QueueHead*> m_free_qh_pool;
+    Vector<TransferDescriptor*> m_free_td_pool;
+    Vector<TransferDescriptor*> m_iso_td_list;
+
+    OwnPtr<QueueHead> m_interrupt_transfer_queue;
+    OwnPtr<QueueHead> m_lowspeed_control_qh;
+    OwnPtr<QueueHead> m_fullspeed_control_qh;
+    OwnPtr<QueueHead> m_bulk_qh;
+    OwnPtr<QueueHead> m_dummy_qh; // Needed for PIIX4 hack
```

Comment:
Is this actually sane? `OwnPtr` is meant for objects that need to be `delete`d. These are allocated via `allocate_queue_head()` and come from a non-heap pool. It doesn't seem safe to `delete` pointers into that pool(?)

Reply:
Yeah these should probably be raw pointers too. I suppose unless explicitly performed otherwise, lifetime of these are guaranteed.